pkg/client/factory.go

```diff
@@ -64,6 +65,7 @@ func (f OptimizelyFactory) Client(clientOptions ...OptionFunc) (*OptimizelyClien
 	}
 
 	appClient := &OptimizelyClient{executionCtx: executionCtx}
+	appClient.NotificationCenter = registry.GetNotificationCenter(f.SDKKey)
```

Comment:
This doesn't need to be set after the object init. You can put it right after the exec context